### PR TITLE
Re-enable sandbox builds in Docker

### PR DESCRIPTION
**Pull request by will-cromar**
Will kick off TPU CI on this branch to confirm the fix

### CODE REVIEW

**Comment by will-cromar** [comment]
The TPU CI and my local build started failing after #4636. Tested these changes locally.

If others are seeing the same issue, we can change the default in `build_torch_xla_libs.sh`

cc @stgpetrovic

File: docker/experimental/Dockerfile [hunk 1]
```diff
@@ -74,7 +74,7 @@ COPY third_party/ third_party/
 COPY build_torch_xla_libs.sh .
 
 # TODO: Remove this when it's not required anymore
-ENV XLA_SANDBOX_BUILD=0
+ENV XLA_SANDBOX_BUILD=1
```
Comment (JackCaoG) [inline docker/experimental/Dockerfile:77]: Do we have plan to remove sandbox in the future?

**Review by JackCaoG — APPROVED** [review]

File: docker/experimental/Dockerfile (hunk 1 above)
Comment (will-cromar) [reply docker/experimental/Dockerfile:77]: I'll defer to @stgpetrovic on that. My understanding is that we need sandboxing to take advantage of remote build execution.

**Comment by will-cromar** [comment]
I'm still getting this error: `ERROR: Config value 'tpu' is not defined in any .rc file`. Any ideas?

**Comment by stgpetrovic** [comment]
The docker file is not copying .bazel{rc,version} nor WORKSPACE so it has
no config.

see the fix in the bazel branch i pushed, i had it running the cloudbuild
test manually before repushing to see if there are additional issues.

cheers,

**Comment by will-cromar** [comment]
Now I'm getting this error after the bazel build starts: `LLVM ERROR: IO failure on output stream: No space left on device`

**Comment by stgpetrovic** [comment]
This is /dev/shm, we can remove that part so it builds slowly on disk, or
make the image have larger shm, ive seen that sonewhere in cloudbuild
configuring the shm size. You can revert and I can debug it all until it
passes manually.

**Comment by will-cromar** [comment]
I'll try moving the TPU CI build to the same pool that we use for the nightly builds so that it will have the same shm size and bazel jobs.

**Review by steventk-g — APPROVED** [review]
Thanks Will!

File: docker/experimental/Dockerfile (hunk 1 above)
Comment (stgpetrovic) [reply docker/experimental/Dockerfile:80]: The sandbox is required by some bug; bazel complains about some llvm mlir deps, that's why the sandbox was introduced before this migration. Once we're running remote, there's no need for sandbox,

Comment (stgpetrovic) [reply docker/experimental/Dockerfile:80]: OK this makes sense: this build was using 8 jobs and was running locally before. We can either leave it at that, or make the RAM a bit bigger and make it much faster.

For now I'd like to fix this so I'll make it run as it did before; with the new .bazelrc we default to shm/sandboxed.

**Comment by will-cromar** [comment]
See #4649

**Pull request closed without merging**